src/indieweb_utils/replies/context.py

```diff
@@ -0,0 +1,326 @@
+            post_video_url = None
+
+            if h_entry["properties"].get("photo"):
+                post_photo_url = canonicalize_url(h_entry["properties"]["photo"][0], domain, url)
+
+            if h_entry["properties"].get("video"):
+                post_video_url = canonicalize_url(h_entry["properties"]["video"][0], domain, url)
+
+            # look for featured image to display in reply context
+            if post_photo_url is None:
+                meta_og_image = soup.find("meta", property="og:image")
+                meta_twitter_image = soup.find("meta", property="twitter:image")
+
+                if meta_og_image and meta_og_image.get("content"):
+                    post_photo_url = meta_og_image["content"]
+
+                elif meta_twitter_image and meta_twitter_image.get("content"):
+                    post_photo_url = meta_twitter_image["content"]
+
+            h_entry = {
+                "type": "entry",
+                "author": {"type": "card", "url": author_url, "name": author_name, "photo": author_image},
+                "name": p_name,
+                "url": url,
+                "content": {"html": post_body},
+            }
+
+            if post_photo_url:
+                h_entry["post_photo_url"] = post_photo_url
+
+            if post_video_url:
+                h_entry["post_video_url"] = post_video_url
+
+            return True, h_entry, site_supports_webmention
+
+        elif parsed["items"] and parsed["items"][0]["type"] == ["h-card"]:
+            h_card = parsed["items"][0]
+
+            if h_card["properties"].get("name"):
+                author_name = h_card["properties"]["name"][0]
+            else:
+                author_name = None
+
+            if h_card["properties"].get("photo"):
+                author_image = h_card["properties"]["photo"][0]
+                if author_image.startswith("//"):
+                    author_image = "https:" + author_image
+                elif author_image.startswith("/"):
+                    author_image = url.split("/")[0] + "//" + domain + author_image
+                elif author_image.startswith("http://") or author_image.startswith("https://"):
+                    author_image = author_image
+                else:
+                    author_image = "https://" + domain + "/" + author_image
+            else:
+                author_image = None
+
+            if h_card["properties"].get("note"):
+                post_body = h_card["properties"]["note"][0]
+            else:
+                post_body = None
+
+            h_entry = {
+                "type": "entry",
+                "author": {"type": "card", "url": url, "name": author_name, "photo": author_image},
+                "url": url,
+                "content": {"html": post_body},
+            }
+
+            return True, h_entry, site_supports_webmention
+
+        h_entry = {}
+
+        if parsed_url.netloc == "twitter.com" and twitter_bearer_token is not None:
```

Comment:
Changed this if statement to use a parsed url, rather than a substring search as CodeQL caught it as a security vulnerability.

Reply:
Good catch! I could not think of an appropriate approach to stop that from happening. Thank you!